src/api/exchange/ExchangeApi.ts

```diff
@@ -104,6 +104,11 @@ export interface ExchangeApi extends DepositApi {
   cancelOrders(params: CancelOrdersParams): Promise<Receipt>
 }
 
+export interface DetailedAuctionElement extends AuctionElement {
+  buyToken: TokenDetails | null
+  sellToken: TokenDetails | null
```

Comment:
What about:

```suggestion
  buyToken?: TokenDetails
  sellToken?: TokenDetails
```

Unless you effectively need `null` for some reason.

Reply:
mm the `getTokenFromExchangeById` fn can return `null`, any other way?

Reply:
Then keep it as it is. `null` is fine.